src/v/pandaproxy/rest/test/consumer_group.cc

```diff
@@ -283,3 +283,186 @@ FIXTURE_TEST(pandaproxy_consumer_group, pandaproxy_test_fixture) {
+            member_id()));
+        BOOST_REQUIRE_EQUAL(
+          res.headers.at(boost::beast::http::field::content_type),
+          to_header_value(ppj::serialization_format::v2));
+    }
+
+    info("Member id: {}", member_id);
+
+    info("Adding known topic");
+    auto tp = model::topic_partition(model::topic("t"), model::partition_id(0));
+    auto ntp = make_default_ntp(tp.topic, tp.partition);
+    add_topic(model::topic_namespace_view(ntp)).get();
+
+    {
+        info("List known topics");
+        auto res = http_request(client, "/topics");
+        BOOST_REQUIRE_EQUAL(
+          res.headers.result(), boost::beast::http::status::ok);
+        BOOST_REQUIRE_EQUAL(res.body, R"(["t"])");
+    }
+    {
+        info("Subscribe consumer");
+        ss::sstring req_body(R"(
+{
+  "topics": [
+    "t"
+  ]
+})");
+        iobuf req_body_buf;
+        req_body_buf.append(req_body.data(), req_body.size());
+        auto res = http_request(
+          client,
+          fmt::format(
+            "/consumers/{}/instances/{}/subscription", group_id(), member_id()),
+          std::move(req_body_buf),
+          boost::beast::http::verb::post,
+          ppj::serialization_format::v2,
+          ppj::serialization_format::v2);
+        BOOST_REQUIRE_EQUAL(
+          res.headers.result(), boost::beast::http::status::no_content);
+    }
+
+    {
+        const ss::sstring post_body(
+          R"({
+"records":[
+    {
+        "key": "cGFuZGFwcm94eQ==",
+        "value": "IlZhbHVlIg=="
+    }
+]
+})");
+
+        info("Produce binary key to known topic");
+        set_client_config("retries", size_t(5));
+        auto body = iobuf();
+        body.append(post_body.data(), post_body.size());
+        auto res = http_request(
+          client,
+          "/topics/t",
+          std::move(body),
+          boost::beast::http::verb::post,
+          ppj::serialization_format::binary_v2,
+          ppj::serialization_format::v2);
+
+        BOOST_REQUIRE_EQUAL(
+          res.headers.result(), boost::beast::http::status::ok);
+        BOOST_REQUIRE_EQUAL(
+          res.body, R"({"offsets":[{"partition":0,"offset":0}]})");
+    }
+
+    {
+        info("Consume from topic");
+        auto res = http_request(
+          client,
+          fmt::format(
+            "/consumers/{}/instances/{}/records?timeout={}&max_bytes={}",
+            group_id(),
+            member_id(),
+            "1000",
+            "1000000"),
+          boost::beast::http::verb::get,
+          ppj::serialization_format::v2,
+          ppj::serialization_format::json_v2);
+        BOOST_REQUIRE_EQUAL(
+          res.headers.result(), boost::beast::http::status::request_timeout);
+        BOOST_REQUIRE_EQUAL(
+          res.body,
+          R"({"error_code":40801,"message":"Unable to serialize key of record at offset 0 in topic:partition t:0"})");
+    }
+
+    {
+        const ss::sstring post_body(
+          R"({
+"records":[
+    {
+        "value": "cGFuZGFwcm94eQ=="
```

Comment:
Isn't this the same test as the previous produce but with no key? If so, is it testing some path of the code that is different?

As I understand it, omitting a key just sets `key=null`; but, the code coverage is essentially the same.

Reply:
There's a different error message if the key or value can't be deserialized.  Lines 53-69 of fetch.h